Bug on Windows/Mac - partially initialized module 'fiona' has no attribute '_loading'

<!--

WELCOME ABOARD!

Hi and welcome to the Fiona project. We appreciate bug reports, questions
about documentation, and suggestions for new features. This issue template
isn't intended to ward you off; only to intercept and redirect one particular
category of reports, and to collect a few important facts that issue reporters
often omit.

You think you've found something? We believe you.

The primary forum for questions about installation and usage of Fiona is 
https://fiona.groups.io/g/main. The authors and other users will answer 
questions when they have expertise to share and time to explain. Please take the
time to craft a clear question and be patient about responses. Please do not
bring these questions to Fiona's issue tracker, which we want to reserve for
bug reports and other actionable issues.

Questions about development of Fiona, brainstorming, requests for comment,
and not-yet-actionable proposals are welcome in the project's developers 
discussion group https://fiona.groups.io/g/dev. Issues opened in Fiona's
GitHub repo which haven't been socialized there may be perfunctorily closed.

Please note: Fiona contains extension modules and is thus susceptible to
C library compatibility issues. If you are reporting an installation or module
import issue, please note that this project only accepts reports about problems
with packages downloaded from the Python Package Index. Conda users should take
issues to one of the following trackers:

- https://github.com/ContinuumIO/anaconda-issues/issues
- https://github.com/conda-forge/fiona-feedstock

-->

## Expected behavior and actual behavior.

When using geopandas, shapley and fiona, the importing of files breaks on Mac OS.
The same problem does not occur on [windows](https://github.com/pypsa-meets-africa/pypsa-africa/runs/5709208035?check_suite_focus=true) or [linux](https://github.com/pypsa-meets-africa/pypsa-africa/runs/5709208030?check_suite_focus=true).
The following error arises:

``` bash
partially initialized module 'fiona' has no attribute '_loading' (most likely due to a circular import)
  File "/Users/runner/work/pypsa-africa/pypsa-africa/Snakefile", line 15, in <module>
  File "/Users/runner/work/pypsa-africa/pypsa-africa/scripts/add_electricity.py", line 93, in <module>
  File "/Users/runner/miniconda3/envs/pypsa-africa/lib/python3.9/site-packages/fiona/__init__.py", line 85, in <module>
ERROR:snakemake.logging:AttributeError in line 15 of /Users/runner/work/pypsa-africa/pypsa-africa/Snakefile:
partially initialized module 'fiona' has no attribute '_loading' (most likely due to a circular import)
  File "/Users/runner/work/pypsa-africa/pypsa-africa/Snakefile", line 15, in <module>
  File "/Users/runner/work/pypsa-africa/pypsa-africa/scripts/add_electricity.py", line 93, in <module>
  File "/Users/runner/miniconda3/envs/pypsa-africa/lib/python3.9/site-packages/fiona/__init__.py", line 85, in <module>
Error: Process completed with exit code 1.
```

## Steps to reproduce the problem.

This is reproducible in the CI shown in: https://github.com/pypsa-meets-africa/pypsa-africa/runs/5709176725?check_suite_focus=true
I attempted various ways, but most likely the errors arise while performing the following series of imports:

``` python
import logging
import os

import geopandas as gpd
import numpy as np
import pandas as pd
import powerplantmatching as pm
import pypsa
import xarray as xr
from powerplantmatching.export import map_country_bus
from shapely.validation import make_valid
```
## Operating system

  Environment: macos-11
  Version: 20220322.1

## Fiona and GDAL version and provenance

Installed using the following environment file: https://github.com/pypsa-meets-africa/pypsa-africa/blob/main/envs/environment.yaml
``` yaml
# packages in environment at /Users/runner/miniconda3/envs/pypsa-africa:
#
# Name                    Version                   Build  Channel
affine                    2.3.1              pyhd8ed1ab_0    conda-forge
amply                     0.1.4                      py_0    conda-forge
anyio                     3.5.0            py39h6e9494a_0    conda-forge
appdirs                   1.4.4              pyh9f0ad1d_0    conda-forge
appnope                   0.1.2            py39h6e9494a_2    conda-forge
argon2-cffi               21.3.0             pyhd8ed1ab_0    conda-forge
argon2-cffi-bindings      21.2.0           py39h89e85a6_1    conda-forge
asttokens                 2.0.5              pyhd8ed1ab_0    conda-forge
atlite                    0.2.5              pyhd8ed1ab_0    conda-forge
attrs                     21.4.0             pyhd8ed1ab_0    conda-forge
babel                     2.9.1              pyh44b3[12](https://github.com/pypsa-meets-africa/pypsa-africa/runs/5709176725?check_suite_focus=true#step:7:12)d_0    conda-forge
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
backports                 1.0                        py_2    conda-forge
backports.functools_lru_cache 1.6.4              pyhd8ed1ab_0    conda-forge
beautifulsoup4            4.10.0             pyha770c72_0    conda-forge
bleach                    4.1.0              pyhd8ed1ab_0    conda-forge
blosc                     1.21.0               he49afe7_0    conda-forge
bokeh                     2.4.2            py39h6e9494a_0    conda-forge
boost-cpp                 1.74.0               hff03dee_4    conda-forge
bottleneck                1.3.4            py39h86b5767_0    conda-forge
branca                    0.4.2              pyhd8ed1ab_0    conda-forge
brotli                    1.0.9                h0d85af4_6    conda-forge
brotli-bin                1.0.9                h0d85af4_6    conda-forge
brotlipy                  0.7.0           py39h89e85a6_1003    conda-forge
bzip2                     1.0.8                h0d85af4_4    conda-forge
c-ares                    1.18.1               h0d85af4_0    conda-forge
ca-certificates           2021.10.8            h033912b_0    conda-forge
cairo                     1.16.0            he43a7df_1008    conda-forge
cartopy                   0.20.1           py39h6878a05_1    conda-forge
cdsapi                    0.5.1              pyhd8ed1ab_0    conda-forge
certifi                   2021.10.8        py39h6e9494a_1    conda-forge
cffi                      1.15.0           py39he338e87_0    conda-forge
cfgv                      3.3.1              pyhd8ed1ab_0    conda-forge
cfitsio                   3.470                h01dc385_7    conda-forge
cftime                    1.6.0            py39h86b5767_0    conda-forge
charset-normalizer        2.0.12             pyhd8ed1ab_0    conda-forge
click                     7.1.2              pyh9f0ad1d_0    conda-forge
click-plugins             1.1.1                      py_0    conda-forge
cligj                     0.7.2              pyhd8ed1ab_1    conda-forge
cloudpickle               2.0.0              pyhd8ed1ab_0    conda-forge
coin-or-cbc               2.10.7               hada2c85_0    conda-forge
coin-or-cgl               0.60.3               h1c2bb20_2    conda-forge
coin-or-clp               1.17.6               h79c0eca_3    conda-forge
coin-or-osi               0.108.6              h009c923_2    conda-forge
coin-or-utils             2.11.6               h07ff368_0    conda-forge
coincbc                   2.10.7            0_metapackage    conda-forge
colorama                  0.4.4              pyh9f0ad1d_0    conda-forge
colorcet                  3.0.0              pyhd8ed1ab_0    conda-forge
conda                     4.12.0           py39h6e9494a_0    conda-forge
conda-package-handling    1.8.0            py39h63b48b0_0    conda-forge
configargparse            1.5.3              pyhd8ed1ab_0    conda-forge
connection_pool           0.0.3              pyhd3deb0d_0    conda-forge
country_converter         0.7.4              pyhd8ed1ab_0    conda-forge
countrycode               0.2                      pypi_0    pypi
cryptography              36.0.2           py39h1644bb1_0    conda-forge
curl                      7.82.0               h9f20792_0    conda-forge
cycler                    0.11.0             pyhd8ed1ab_0    conda-forge
cytoolz                   0.11.2           py39h89e85a6_1    conda-forge
dask                      2021.3.1           pyhd8ed1ab_0    conda-forge
dask-core                 2021.3.1           pyhd8ed1ab_0    conda-forge
datashader                0.[13](https://github.com/pypsa-meets-africa/pypsa-africa/runs/5709176725?check_suite_focus=true#step:7:13).0             pyh6c4a22f_0    conda-forge
datashape                 0.5.4                      py_1    conda-forge
datrie                    0.8.2            py39h89e85a6_3    conda-forge
debugpy                   1.5.1            py39h9fcab8e_0    conda-forge
decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
deprecation               2.1.0              pyh9f0ad1d_0    conda-forge
descartes                 1.1.0                      py_4    conda-forge
distlib                   0.3.4              pyhd8ed1ab_0    conda-forge
distributed               2021.4.1         py39h6e9494a_1    conda-forge
docutils                  0.18.1           py39h6e9494a_0    conda-forge
entrypoints               0.4                pyhd8ed1ab_0    conda-forge
entsoe-py                 0.5.2                    pypi_0    pypi
esy-osm-pbf               0.0.1                    pypi_0    pypi
esy-osmfilter             1.0.11                   pypi_0    pypi
et_xmlfile                1.0.1                   py_1001    conda-forge
executing                 0.8.3              pyhd8ed1ab_0    conda-forge
expat                     2.4.7                h96cf925_0    conda-forge
filelock                  3.6.0              pyhd8ed1ab_0    conda-forge
fiona                     1.8.20           py39h10412eb_2    conda-forge
flit-core                 3.7.1              pyhd8ed1ab_0    conda-forge
folium                    0.12.1.post1       pyhd8ed1ab_1    conda-forge
font-ttf-dejavu-sans-mono 2.37                 hab24e00_0    conda-forge
font-ttf-inconsolata      3.000                h77eed37_0    conda-forge
font-ttf-source-code-pro  2.038                h77eed37_0    conda-forge
font-ttf-ubuntu           0.83                 hab24e00_0    conda-forge
fontconfig                2.13.96              h676cef8_2    conda-forge
fonts-conda-ecosystem     1                             0    conda-forge
fonts-conda-forge         1                             0    conda-forge
fonttools                 4.31.2           py39h63b48b0_0    conda-forge
freetype                  2.10.4               h4cff582_1    conda-forge
freexl                    1.0.6                h0d85af4_0    conda-forge
fsspec                    2022.2.0           pyhd8ed1ab_0    conda-forge
gdal                      3.3.3            py39he7e3f19_0    conda-forge
geographiclib             1.52               pyhd8ed1ab_0    conda-forge
geopandas                 0.10.2             pyhd8ed1ab_1    conda-forge
geopandas-base            0.10.2             pyha770c72_1    conda-forge
geopy                     2.2.0              pyhd8ed1ab_0    conda-forge
geos                      3.9.1                he49afe7_2    conda-forge
geotiff                   1.7.0                hb0faedc_3    conda-forge
geoviews                  1.9.5              pyhd8ed1ab_0    conda-forge
geoviews-core             1.9.5              pyha770c72_0    conda-forge
gettext                   0.19.8.1          hd1a6beb_1008    conda-forge
giflib                    5.2.1                hbcb3906_2    conda-forge
gitdb                     4.0.9              pyhd8ed1ab_0    conda-forge
gitpython                 3.1.27             pyhd8ed1ab_0    conda-forge
glpk                      5.0                  h3cb5acd_0    conda-forge
gmp                       6.2.1                h2e338ed_0    conda-forge
googledrivedownloader     0.4                pyhd3deb0d_1    conda-forge
hdf4                      4.2.15               hefd3b78_3    conda-forge
hdf5                      1.12.1          nompi_ha60fbc9_104    conda-forge
heapdict                  1.0.1                      py_0    conda-forge
holoviews                 1.[14](https://github.com/pypsa-meets-africa/pypsa-africa/runs/5709176725?check_suite_focus=true#step:7:14).8             pyhd8ed1ab_0    conda-forge
hvplot                    0.7.3              pyh6c4a22f_0    conda-forge
icu                       68.2                 he49afe7_0    conda-forge
identify                  2.4.12             pyhd8ed1ab_0    conda-forge
idna                      3.3                pyhd8ed1ab_0    conda-forge
importlib-metadata        4.11.3           py39h6e9494a_0    conda-forge
importlib_metadata        4.11.3               hd8ed1ab_0    conda-forge
importlib_resources       5.6.0              pyhd8ed1ab_0    conda-forge
iniconfig                 1.1.1              pyh9f0ad1d_0    conda-forge
ipopt                     3.13.2               h654a6ca_1    conda-forge
ipykernel                 6.9.2            py39h71a6800_0    conda-forge
ipython                   8.1.1            py39h6e9494a_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
jbig                      2.1               h0d85af4_2003    conda-forge
jedi                      0.18.1           py39h6e9494a_0    conda-forge
jinja2                    3.1.1              pyhd8ed1ab_0    conda-forge
joblib                    1.1.0              pyhd8ed1ab_0    conda-forge
jpeg                      9e                   h0d85af4_0    conda-forge
json-c                    0.[15](https://github.com/pypsa-meets-africa/pypsa-africa/runs/5709176725?check_suite_focus=true#step:7:15)                 hcb556a6_0    conda-forge
json5                     0.9.5              pyh9f0ad1d_0    conda-forge
jsonschema                4.4.0              pyhd8ed1ab_0    conda-forge
jupyter_client            7.1.2              pyhd8ed1ab_0    conda-forge
jupyter_core              4.9.2            py39h6e9494a_0    conda-forge
jupyter_server            1.15.6             pyhd8ed1ab_1    conda-forge
jupyterlab                3.3.2              pyhd8ed1ab_0    conda-forge
jupyterlab_pygments       0.1.2              pyh9f0ad1d_0    conda-forge
jupyterlab_server         2.11.0             pyhd8ed1ab_0    conda-forge
kealib                    1.4.14               ha22a8b1_3    conda-forge
kiwisolver                1.4.0            py39h7248d28_0    conda-forge
krb5                      1.19.3               hb49756b_0    conda-forge
lcms2                     2.12                 h577c468_0    conda-forge
lerc                      3.0                  he49afe7_0    conda-forge
libarchive                3.5.2                h2b60450_1    conda-forge
libblas                   3.9.0           13_osx64_openblas    conda-forge
libbrotlicommon           1.0.9                h0d85af4_6    conda-forge
libbrotlidec              1.0.9                h0d85af4_6    conda-forge
libbrotlienc              1.0.9                h0d85af4_6    conda-forge
libcblas                  3.9.0           13_osx64_openblas    conda-forge
libcurl                   7.82.0               h9f20792_0    conda-forge
libcxx                    13.0.1               hc203e6f_0    conda-forge
libdap4                   3.20.6               h3e144a0_2    conda-forge
libdeflate                1.10                 h0d85af4_0    conda-forge
libedit                   3.1.20191231         h0678c8f_2    conda-forge
libev                     4.33                 haf1e3a3_1    conda-forge
libffi                    3.4.2                h0d85af4_5    conda-forge
libgdal                   3.3.3                h053b6da_0    conda-forge
libgfortran               5.0.0           9_3_0_h6c81a4c_23    conda-forge
libgfortran5              9.3.0               h6c81a4c_23    conda-forge
libglib                   2.70.2               hf1fb8c0_4    conda-forge
libiconv                  1.[16](https://github.com/pypsa-meets-africa/pypsa-africa/runs/5709176725?check_suite_focus=true#step:7:16)                 haf1e3a3_0    conda-forge
libkml                    1.3.0             h8fd9edb_1014    conda-forge
liblapack                 3.9.0           13_osx64_openblas    conda-forge
liblapacke                3.9.0           13_osx64_openblas    conda-forge
libllvm10                 10.0.1               h009f743_3    conda-forge
libmamba                  0.22.1               hf91a703_0    conda-forge
libmambapy                0.22.1           py39h434a9b6_0    conda-forge
libnetcdf                 4.8.1           nompi_h6609ca0_101    conda-forge
libnghttp2                1.47.0               h942079c_0    conda-forge
libopenblas               0.3.18          openmp_h3351f45_0    conda-forge
libpng                    1.6.37               h7cec526_2    conda-forge
libpq                     13.5                 hea3049e_1    conda-forge
librttopo                 1.1.0                h5413771_6    conda-forge
libsodium                 1.0.18               hbcb3906_1    conda-forge
libsolv                   0.7.19               hcf210ce_5    conda-forge
libspatialindex           1.9.3                he49afe7_4    conda-forge
libspatialite             5.0.1                hfc891e7_9    conda-forge
libssh2                   1.10.0               h52ee1ee_2    conda-forge
libtiff                   4.3.0                h[17](https://github.com/pypsa-meets-africa/pypsa-africa/runs/5709176725?check_suite_focus=true#step:7:17)f2ce3_3    conda-forge
libwebp                   1.2.2                h28dabe5_0    conda-forge
libwebp-base              1.2.2                h0d85af4_1    conda-forge
libxcb                    1.13              h0d85af4_1004    conda-forge
libxml2                   2.9.12               h93ec3fd_0    conda-forge
libxslt                   1.1.33               h5739fc3_2    conda-forge
libzip                    1.8.0                h8b0c345_1    conda-forge
libzlib                   1.2.11            h6c3fc93_1014    conda-forge
llvm-openmp               13.0.1               hcb1a161_1    conda-forge
llvmlite                  0.36.0           py39h798a4f4_0    conda-forge
locket                    0.2.0                      py_2    conda-forge
lxml                      4.8.0            py39h63b48b0_1    conda-forge
lz4-c                     1.9.3                he49afe7_1    conda-forge
lzo                       2.10              haf1e3a3_1000    conda-forge
mamba                     0.22.1           py39ha435c47_0    conda-forge
mapclassify               2.4.3              pyhd8ed1ab_0    conda-forge
markdown                  3.3.6              pyhd8ed1ab_0    conda-forge
markupsafe                2.1.1            py39h63b48b0_0    conda-forge
matplotlib                3.5.1            py39h6e9494a_0    conda-forge
matplotlib-base           3.5.1            py39hb07454d_0    conda-forge
matplotlib-inline         0.1.3              pyhd8ed1ab_0    conda-forge
memory_profiler           0.60.0             pyhd8ed1ab_0    conda-forge
metis                     5.1.0             h2e338ed_1006    conda-forge
mistune                   0.8.4           py39h89e85a6_1005    conda-forge
msgpack-python            1.0.3            py39hf0[18](https://github.com/pypsa-meets-africa/pypsa-africa/runs/5709176725?check_suite_focus=true#step:7:18)cea_0    conda-forge
multipledispatch          0.6.0                      py_0    conda-forge
munch                     2.5.0                      py_0    conda-forge
munkres                   1.1.4              pyh9f0ad1d_0    conda-forge
nbclassic                 0.3.7              pyhd8ed1ab_0    conda-forge
nbclient                  0.5.13             pyhd8ed1ab_0    conda-forge
nbconvert                 6.4.4            py39h6e9494a_0    conda-forge
nbformat                  5.2.0              pyhd8ed1ab_0    conda-forge
ncurses                   6.3                  he49afe7_0    conda-forge
nest-asyncio              1.5.4              pyhd8ed1ab_0    conda-forge
netcdf4                   1.5.8           nompi_py39he7d1c46_101    conda-forge
networkx                  2.7.1              pyhd8ed1ab_1    conda-forge
nodeenv                   1.6.0              pyhd8ed1ab_0    conda-forge
notebook                  6.4.10             pyha770c72_0    conda-forge
notebook-shim             0.1.0              pyhd8ed1ab_0    conda-forge
nspr                      4.32                 hcd9eead_1    conda-forge
nss                       3.76                 hfce436b_0    conda-forge
numba                     0.53.1           py39h32e38f5_1    conda-forge
numexpr                   2.8.0            py39h4d6be9b_1    conda-forge
numpy                     1.22.3           py39hf56e92f_0    conda-forge
openjpeg                  2.4.0                h6e7aa92_1    conda-forge
openpyxl                  3.0.9              pyhd8ed1ab_0    conda-forge
openssl                   1.1.1n               h6c3fc93_0    conda-forge
packaging                 21.3               pyhd8ed1ab_0    conda-forge
pandas                    1.4.1                    pypi_0    pypi
pandoc                    2.17.1.1             h694c41f_0    conda-forge
pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
panel                     0.12.6             pyhd8ed1ab_0    conda-forge
param                     1.12.0             pyh6c4a22f_0    conda-forge
parso                     0.8.3              pyhd8ed1ab_0    conda-forge
partd                     1.2.0              pyhd8ed1ab_0    conda-forge
patsy                     0.5.2              pyhd8ed1ab_0    conda-forge
pcre                      8.45                 he49afe7_0    conda-forge
pexpect                   4.8.0              pyh9f0ad1d_2    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pillow                    9.0.1            py39hd2c7aa1_2    conda-forge
pip                       22.0.4             pyhd8ed1ab_0    conda-forge
pixman                    0.40.0               hbcb3906_0    conda-forge
plac                      1.3.4              pyhd8ed1ab_0    conda-forge
platformdirs              2.5.1              pyhd8ed1ab_0    conda-forge
pluggy                    1.0.0            py39h6e9494a_2    conda-forge
ply                       3.11                       py_1    conda-forge
poppler                   21.09.0              h9573804_3    conda-forge
poppler-data              0.4.11               hd8ed1ab_0    conda-forge
postgresql                13.5                 he8fe76e_1    conda-forge
powerplantmatching        0.4.8                    pypi_0    pypi
pre-commit                2.17.0           py39h6e9494a_0    conda-forge
progressbar2              4.0.0              pyhd8ed1ab_0    conda-forge
proj                      8.1.1                h1512c50_2    conda-forge
prometheus_client         0.13.1             pyhd8ed1ab_0    conda-forge
prompt-toolkit            3.0.27             pyha770c72_0    conda-forge
protobuf                  3.[19](https://github.com/pypsa-meets-africa/pypsa-africa/runs/5709176725?check_suite_focus=true#step:7:19).4                   pypi_0    pypi
psutil                    5.9.0            py39h89e85a6_0    conda-forge
pthread-stubs             0.4               hc929b4f_1001    conda-forge
ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
pulp                      2.6.0            py39h6e9494a_0    conda-forge
pure_eval                 0.2.2              pyhd8ed1ab_0    conda-forge
py                        1.11.0             pyh6c4a22f_0    conda-forge
pybind11-abi              4                    hd8ed1ab_3    conda-forge
pycosat                   0.6.3           py39h89e85a6_1009    conda-forge
pycountry                 [20](https://github.com/pypsa-meets-africa/pypsa-africa/runs/5709176725?check_suite_focus=true#step:7:20).7.3             pyh9f0ad1d_0    conda-forge
pycparser                 2.[21](https://github.com/pypsa-meets-africa/pypsa-africa/runs/5709176725?check_suite_focus=true#step:7:21)               pyhd8ed1ab_0    conda-forge
pyct                      0.4.6                      py_0    conda-forge
pyct-core                 0.4.6                      py_0    conda-forge
pygments                  2.11.2             pyhd8ed1ab_0    conda-forge
pyomo                     6.4.0            py39hfd1d529_0    conda-forge
pyopenssl                 [22](https://github.com/pypsa-meets-africa/pypsa-africa/runs/5709176725?check_suite_focus=true#step:7:22).0.0             pyhd8ed1ab_0    conda-forge
pyparsing                 3.0.7              pyhd8ed1ab_0    conda-forge
pyproj                    3.2.1            py39h5ea97c7_2    conda-forge
pypsa                     0.19.2             pyhd8ed1ab_0    conda-forge
pyrsistent                0.18.1           py39h89e85a6_0    conda-forge
pyshp                     2.2.0              pyhd8ed1ab_0    conda-forge
pysocks                   1.7.1            py39h6e9494a_4    conda-forge
pytables                  3.7.0            py39hfd850c7_0    conda-forge
pytest                    7.1.1            py39h6e9494a_0    conda-forge
python                    3.9.12          h8b4d769_1_cpython    conda-forge
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python-utils              3.1.0              pyhd8ed1ab_0    conda-forge
python_abi                3.9                      2_cp39    conda-forge
pytz                      2022.1             pyhd8ed1ab_0    conda-forge
pyviz_comms               2.1.0              pyhd8ed1ab_0    conda-forge
pyyaml                    6.0              py39h89e85a6_3    conda-forge
pyzmq                     22.3.0           py39h7fec2f1_1    conda-forge
rasterio                  1.2.10           py39hf4c83f0_0    conda-forge
ratelimiter               1.2.0                   py_1002    conda-forge
readline                  8.1                  h05e3726_0    conda-forge
reproc                    14.2.3               h0d85af4_0    conda-forge
reproc-cpp                14.2.3               he49afe7_0    conda-forge
requests                  2.27.1             pyhd8ed1ab_0    conda-forge
retry                     0.9.2                      py_0    conda-forge
rioxarray                 0.10.2             pyhd8ed1ab_0    conda-forge
rtree                     0.9.7            py39h7d0d40a_3    conda-forge
ruamel_yaml               0.15.80         py39h89e85a6_1006    conda-forge
scikit-learn              1.0.2            py39hd4eea88_0    conda-forge
scipy                     1.8.0            py39h056f1c0_1    conda-forge
seaborn                   0.11.2               hd8ed1ab_0    conda-forge
seaborn-base              0.11.2             pyhd8ed1ab_0    conda-forge
send2trash                1.8.0              pyhd8ed1ab_0    conda-forge
setuptools                59.8.0           py39h6e9494a_0    conda-forge
shapely                   1.8rc1           py39h580b1cf_0    conda-forge/label/shapely_dev
six                       1.16.0             pyh6c4a22f_0    conda-forge
sklearn                   0.0                      pypi_0    pypi
smart_open                5.2.1              pyhd8ed1ab_0    conda-forge
smmap                     3.0.5              pyh44b312d_0    conda-forge
snakemake-minimal         7.3.2              pyhdfd78af_0    bioconda
sniffio                   1.2.0            py39h6e9494a_2    conda-forge
snuggs                    1.4.7                      py_0    conda-forge
sortedcontainers          2.4.0              pyhd8ed1ab_0    conda-forge
soupsieve                 2.3.1              pyhd8ed1ab_0    conda-forge
sqlite                    3.37.1               hb516[25](https://github.com/pypsa-meets-africa/pypsa-africa/runs/5709176725?check_suite_focus=true#step:7:25)3_0    conda-forge
stack_data                0.2.0              pyhd8ed1ab_0    conda-forge
statsmodels               0.13.2           py39hc89836e_0    conda-forge
stopit                    1.1.2                      py_0    conda-forge
tabulate                  0.8.9              pyhd8ed1ab_0    conda-forge
tblib                     1.7.0              pyhd8ed1ab_0    conda-forge
terminado                 0.13.3           py39h6e9494a_0    conda-forge
testpath                  0.6.0              pyhd8ed1ab_0    conda-forge
threadpoolctl             3.1.0              pyh8a188c0_0    conda-forge
tiledb                    2.3.4                h8370e7a_0    conda-forge
tk                        8.6.12               h5dbffcc_0    conda-forge
toml                      0.10.2             pyhd8ed1ab_0    conda-forge
tomli                     2.0.1              pyhd8ed1ab_0    conda-forge
toolz                     0.11.2             pyhd8ed1ab_0    conda-forge
toposort                  1.7                pyhd8ed1ab_0    conda-forge
tornado                   6.1              py39h89e85a6_2    conda-forge
tqdm                      4.63.1             pyhd8ed1ab_0    conda-forge
traitlets                 5.1.1              pyhd8ed1ab_0    conda-forge
tsam                      2.0.0                    pypi_0    pypi
typing_extensions         4.1.1              pyha770c72_0    conda-forge
tzcode                    2022a                h5eb16cf_0    conda-forge
tzdata                    2022a                h191b570_0    conda-forge
ukkonen                   1.0.1            py39hf018cea_1    conda-forge
unicodedata2              14.0.0           py39h89e85a6_0    conda-forge
urllib3                   1.[26](https://github.com/pypsa-meets-africa/pypsa-africa/runs/5709176725?check_suite_focus=true#step:7:26).9             pyhd8ed1ab_0    conda-forge
virtualenv                20.14.0          py39h6e9494a_0    conda-forge
vresutils                 0.3.1                    pypi_0    pypi
wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
webencodings              0.5.1                      py_1    conda-forge
websocket-client          1.3.1              pyhd8ed1ab_0    conda-forge
wheel                     0.37.1             pyhd8ed1ab_0    conda-forge
wrapt                     1.14.0           py39h63b48b0_0    conda-forge
xarray                    2022.3.1.dev[27](https://github.com/pypsa-meets-africa/pypsa-africa/runs/5709176725?check_suite_focus=true#step:7:27)+g8f42bfd3          pypi_0    pypi
xerces-c                  3.2.3                h379762d_3    conda-forge
xlrd                      2.0.1              pyhd8ed1ab_3    conda-forge
xorg-libxau               1.0.9                h35c211d_0    conda-forge
xorg-libxdmcp             1.1.3                h35c211d_0    conda-forge
xyzservices               2022.3.0           pyhd8ed1ab_0    conda-forge
xz                        5.2.5                haf1e3a3_1    conda-forge
yaml                      0.2.5                h0d85af4_2    conda-forge
yaml-cpp                  0.6.3                hb1e8[31](https://github.com/pypsa-meets-africa/pypsa-africa/runs/5709176725?check_suite_focus=true#step:7:31)3_4    conda-forge
yte                       1.2.0            py[39](https://github.com/pypsa-meets-africa/pypsa-africa/runs/5709176725?check_suite_focus=true#step:7:39)h6e9[49](https://github.com/pypsa-meets-africa/pypsa-africa/runs/5709176725?check_suite_focus=true#step:7:49)4a_0    conda-forge
zeromq                    4.3.4                he49afe7_1    conda-forge
zict                      2.1.0              pyhd8ed1ab_0    conda-forge
zipp                      3.7.0              pyhd8ed1ab_1    conda-forge
zlib                      1.2.11            h6c3fc93_1014    conda-forge
zstd                      1.5.2                h[58](https://github.com/pypsa-meets-africa/pypsa-africa/runs/5709176725?check_suite_focus=true#step:7:58)2d3a0_0    conda-forge
```

--- a/CHANGES.txt
+++ b/CHANGES.txt
@@ -8,6 +8,8 @@ All issue numbers are relative to https://github.com/Toblerity/Fiona/issues.
 
 Changes:
 
+* The fiona._loading module, which supports DLL loading on Windows,
+  has been moved into __init__.py and is no longer used anywhere else (#1168).
 * Move project metadata to pyproject.toml (#1165).
 * Update drvsupport.py to reflect new format capabilities in GDAL 3.6.0
   (#1122).

--- a/fiona/__init__.py
+++ b/fiona/__init__.py
@@ -76,42 +76,45 @@ except ImportError:  # pragma: no cover
     class Path:
         pass
 
-# TODO: remove this? Or at least move it, flake8 complains.
-if sys.platform == "win32":
-    libdir = os.path.join(os.path.dirname(__file__), ".libs")
-    os.environ["PATH"] = os.environ["PATH"] + ";" + libdir
+if platform.system() == "Windows":
+    _whl_dir = os.path.join(os.path.dirname(__file__), ".libs")
+    if os.path.exists(_whl_dir):
+        os.add_dll_directory(_whl_dir)
+    else:
+        if "PATH" in os.environ:
+            for p in os.environ["PATH"].split(os.pathsep):
+                if glob.glob(os.path.join(p, "gdal*.dll")):
+                    os.add_dll_directory(p)
 
-import fiona._loading
 
-with fiona._loading.add_gdal_dll_directories():
-    from fiona.collection import BytesCollection, Collection
-    from fiona.drvsupport import supported_drivers
-    from fiona.env import ensure_env_with_credentials, Env
-    from fiona.errors import FionaDeprecationWarning
-    from fiona._env import driver_count
-    from fiona._env import (
-        calc_gdal_version_num,
-        get_gdal_version_num,
-        get_gdal_release_name,
-        get_gdal_version_tuple,
-    )
-    from fiona.io import MemoryFile
-    from fiona.ogrext import (
-        _bounds,
-        _listlayers,
-        _listdir,
-        FIELD_TYPES_MAP,
-        _remove,
-        _remove_layer,
-    )
-    from fiona.path import ParsedPath, parse_path, vsi_path
-    from fiona.vfs import parse_paths as vfs_parse_paths
-    from fiona._show_versions import show_versions
+from fiona.collection import BytesCollection, Collection
+from fiona.drvsupport import supported_drivers
+from fiona.env import ensure_env_with_credentials, Env
+from fiona.errors import FionaDeprecationWarning
+from fiona._env import driver_count
+from fiona._env import (
+    calc_gdal_version_num,
+    get_gdal_version_num,
+    get_gdal_release_name,
+    get_gdal_version_tuple,
+)
+from fiona.io import MemoryFile
+from fiona.ogrext import (
+    _bounds,
+    _listlayers,
+    _listdir,
+    FIELD_TYPES_MAP,
+    _remove,
+    _remove_layer,
+)
+from fiona.path import ParsedPath, parse_path, vsi_path
+from fiona.vfs import parse_paths as vfs_parse_paths
+from fiona._show_versions import show_versions
 
-    # These modules are imported by fiona.ogrext, but are also import here to
-    # help tools like cx_Freeze find them automatically
-    from fiona import _geometry, _err, rfc3339
-    import uuid
+# These modules are imported by fiona.ogrext, but are also import here to
+# help tools like cx_Freeze find them automatically
+from fiona import _geometry, _err, rfc3339
+import uuid
 
 
 __all__ = ['bounds', 'listlayers', 'listdir', 'open', 'prop_type', 'prop_width']

--- a/fiona/collection.py
+++ b/fiona/collection.py
@@ -8,34 +8,31 @@ import os
 import warnings
 from collections import OrderedDict
 
-import fiona._loading
-
-with fiona._loading.add_gdal_dll_directories():
-    from fiona import compat, vfs
-    from fiona.ogrext import Iterator, ItemsIterator, KeysIterator
-    from fiona.ogrext import Session, WritingSession
-    from fiona.ogrext import buffer_to_virtual_file, remove_virtual_file, GEOMETRY_TYPES
-    from fiona.errors import (
-        DriverError,
-        SchemaError,
-        CRSError,
-        UnsupportedGeometryTypeError,
-        UnsupportedOperation,
-        DriverSupportError,
-    )
-    from fiona.logutils import FieldSkipLogFilter
-    from fiona.crs import CRS
-    from fiona._env import get_gdal_release_name, get_gdal_version_tuple
-    from fiona.env import env_ctx_if_needed
-    from fiona.errors import FionaDeprecationWarning
-    from fiona.drvsupport import (
-        driver_from_extension,
-        supported_drivers,
-        driver_mode_mingdal,
-        _driver_converts_field_type_silently_to_str,
-        _driver_supports_field,
-    )
-    from fiona.path import Path, vsi_path, parse_path
+from fiona import compat, vfs
+from fiona.ogrext import Iterator, ItemsIterator, KeysIterator
+from fiona.ogrext import Session, WritingSession
+from fiona.ogrext import buffer_to_virtual_file, remove_virtual_file, GEOMETRY_TYPES
+from fiona.errors import (
+    DriverError,
+    SchemaError,
+    CRSError,
+    UnsupportedGeometryTypeError,
+    UnsupportedOperation,
+    DriverSupportError,
+)
+from fiona.logutils import FieldSkipLogFilter
+from fiona.crs import CRS
+from fiona._env import get_gdal_release_name, get_gdal_version_tuple
+from fiona.env import env_ctx_if_needed
+from fiona.errors import FionaDeprecationWarning
+from fiona.drvsupport import (
+    driver_from_extension,
+    supported_drivers,
+    driver_mode_mingdal,
+    _driver_converts_field_type_silently_to_str,
+    _driver_supports_field,
+)
+from fiona.path import Path, vsi_path, parse_path
 
 
 _GDAL_VERSION_TUPLE = get_gdal_version_tuple()

--- a/fiona/env.py
+++ b/fiona/env.py
@@ -10,22 +10,19 @@ import warnings
 
 import attr
 
-import fiona._loading
-
-with fiona._loading.add_gdal_dll_directories():
-    from fiona._env import (
-        GDALDataFinder,
-        GDALEnv,
-        PROJDataFinder,
-        calc_gdal_version_num,
-        get_gdal_config,
-        get_gdal_release_name,
-        get_gdal_version_num,
-        set_gdal_config,
-        set_proj_data_search_path,
-    )
-    from fiona.errors import EnvError, FionaDeprecationWarning, GDALVersionError
-    from fiona.session import Session, DummySession
+from fiona._env import (
+    GDALDataFinder,
+    GDALEnv,
+    PROJDataFinder,
+    calc_gdal_version_num,
+    get_gdal_config,
+    get_gdal_release_name,
+    get_gdal_version_num,
+    set_gdal_config,
+    set_proj_data_search_path,
+)
+from fiona.errors import EnvError, FionaDeprecationWarning, GDALVersionError
+from fiona.session import Session, DummySession
 
 
 class ThreadEnv(threading.local):

--- a/fiona/fio/env.py
+++ b/fiona/fio/env.py
@@ -6,9 +6,7 @@ import os
 import click
 
 import fiona
-import fiona._loading
-with fiona._loading.add_gdal_dll_directories():
-    from fiona._env import GDALDataFinder, PROJDataFinder
+from fiona._env import GDALDataFinder, PROJDataFinder
 
 
 @click.command(short_help="Print information about the fio environment.")

--- a/fiona/io.py
+++ b/fiona/io.py
@@ -3,12 +3,11 @@
 
 from collections import OrderedDict
 import logging
-import fiona._loading
-with fiona._loading.add_gdal_dll_directories():
-    from fiona.ogrext import MemoryFileBase
-    from fiona.collection import Collection
-    from fiona.meta import supports_vsi
-    from fiona.errors import DriverError
+
+from fiona.ogrext import MemoryFileBase
+from fiona.collection import Collection
+from fiona.meta import supports_vsi
+from fiona.errors import DriverError
 
 log = logging.getLogger(__name__)
 

--- a/fiona/meta.py
+++ b/fiona/meta.py
@@ -1,9 +1,9 @@
-import xml.etree.ElementTree as ET
 import logging
+import xml.etree.ElementTree as ET
+
 import fiona
-with fiona._loading.add_gdal_dll_directories():
-    from fiona.ogrext import _get_metadata_item
-    from fiona.env import require_gdal_version
+from fiona.env import require_gdal_version
+from fiona.ogrext import _get_metadata_item
 
 log = logging.getLogger(__name__)
 

--- a/fiona/session.py
+++ b/fiona/session.py
@@ -3,10 +3,7 @@
 import logging
 import os
 
-import fiona._loading
-
-with fiona._loading.add_gdal_dll_directories():
-    from fiona.path import parse_path, UnparsedPath
+from fiona.path import parse_path, UnparsedPath
 
 log = logging.getLogger(__name__)
 

--- a/fiona/transform.py
+++ b/fiona/transform.py
@@ -2,13 +2,10 @@
 
 from warnings import warn
 
-import fiona._loading
-
-with fiona._loading.add_gdal_dll_directories():
-    from fiona._transform import _transform, _transform_geom
-    from fiona.compat import DICT_TYPES
-    from fiona.errors import FionaDeprecationWarning
-    from fiona.model import _guard_model_object, Geometry
+from fiona._transform import _transform, _transform_geom
+from fiona.compat import DICT_TYPES
+from fiona.errors import FionaDeprecationWarning
+from fiona.model import _guard_model_object, Geometry
 
 
 def transform(src_crs, dst_crs, xs, ys):

